tests/e2e/scale/test_pvc_creation_deletion_scale.py

```diff
@@ -69,192 +80,171 @@ def test_multiple_pvc_creation_deletion_scale(self, namespace, access_mode, inte
         )
 
         # Check for PVC status using threads
-        threads = list()
-        for obj in pvc_objs:
-            process = threading.Thread(
-                target=helpers.wait_for_resource_state,
-                args=(obj, constants.STATUS_BOUND, )
-            )
-            process.start()
-            threads.append(process)
-        for process in threads:
-            process.join()
-
+        helpers.handle_threading(
+            self.pvc_objs,
+            (lambda x: helpers.wait_for_resource_state),
+            (lambda x: (x, constants.STATUS_BOUND))
+        )
         # Get pvc_name, require pvc_name to fetch creation time data from log
+        helpers.handle_threading(
+            self.pvc_objs,
+            (lambda x: x.reload)
+        )
+        pvc_nm_list, self.pv_name_list = ([] for i in range(2))
         threads = list()
-        for pvc_obj in pvc_objs:
-            process = threading.Thread(target=pvc_obj.reload)
-            process.start()
-            threads.append(process)
-        for process in threads:
-            process.join()
-
-        pvc_name_list, pv_name_list = ([] for i in range(2))
-        threads = list()
-        for pvc_obj in pvc_objs:
-            process1 = threading.Thread(target=pvc_name_list.append(pvc_obj.name))
-            process2 = threading.Thread(target=pv_name_list.append(pvc_obj.backed_pv))
+        for pvc_obj in self.pvc_objs:
+            process1 = threading.Thread(target=pvc_nm_list.append(pvc_obj.name))
+            process2 = threading.Thread(target=self.pv_name_list.append(pvc_obj.backed_pv))
             process1.start()
             process2.start()
             threads.append(process1)
             threads.append(process2)
         for process in threads:
             process.join()
+        self.pvc_nm_list = [x.name for x in self.pvc_objs]
+        self.pv_nm_list = [x.backed_pv for x in self.pvc_objs]
+        helpers.handle_threading(
+            self.pvc_objs,
+            (lambda x: x.name)
+        )
+        helpers.handle_threading(
+            self.pvc_objs,
+            (lambda x: x.backed_pv)
+        )
 
         # Get PVC creation time
         pvc_create_time = helpers.measure_pvc_creation_time_bulk(
```

Comment:
@wusui in measure_pvc_creation_time_bulk function we need to add a check either the respective log available or not.
In recent Automation run we got issues like “IndexError: list index out of range” please find respective log in https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/8225/
Better to add exception and failure message if the list is empty.

Reply:
Okay.